### PR TITLE
[4.2.x] Add versioning support for API Products

### PR DESCRIPTION
**Pull request by SavinduDimal**
## Purpose

- Add versioning support for API Products
- Backward port https://github.com/wso2/product-apim-tooling/pull/1022
- Related issues: https://github.com/wso2-enterprise/wso2-apim-internal/issues/4823

### CODE REVIEW

File: import-export-cli/git/gitUtils.go [hunk 1]
```diff
@@ -388,9 +388,12 @@ func handleIfError(err error, failedProjects map[string][]*params.ProjectParams,
 // updatedProjectsPerType is a map of string -> ProjectParams which consists of updated projects per each type (API, App..)
 // Returns bool, true if any deleted projects exists so the process should continue with project deletion path
 // Returns map[string][]*params.ProjectParams, a map of project type (API, App.. ) to each project detail which are
-//  deleted projects
+//
```
Comment (npamudika) [inline import-export-cli/git/gitUtils.go:391]: Seems this change is not required. Please check.

File: import-export-cli/git/gitUtils.go [hunk 2]
```diff
@@ -388,9 +388,12 @@ func handleIfError(err error, failedProjects map[string][]*params.ProjectParams,
...
+//	deleted projects
+//
 // Returns map[string][]*params.ProjectParams, a map of project type (API, App.. ) to each project detail which are
-//  failed during the deployment
+//
```
Comment (npamudika) [inline import-export-cli/git/gitUtils.go:395]: Seems this change is not required. Please check.

File: import-export-cli/git/gitUtils.go [hunk 3]
```diff
@@ -629,8 +632,9 @@ func DeployChangedFiles(accessToken, environment string) map[string][]*params.Pr
 }
 
 // Create 'vcs.yaml' in the repository root folder with a unique id (uuid) for the repository.
-//  If the value of force is false, and the file is already created, gives an error.
-//  If the value of force is true, It will reinitialize the file even if it already exists.
+//
+//	If the value of force is false, and the file is already created, gives an error.
+//	If the value of force is true, It will reinitialize the file even if it already exists.
```
Comment (npamudika) [inline import-export-cli/git/gitUtils.go:637]: Seems this change is not required. Please check.

File: import-export-cli/git/gitUtils.go [hunk 4]
```diff
@@ -700,7 +704,9 @@ func logChangedFiles(changedFileList []string) {
 }
 
 // subPath denotes a single changed file retrieved from "git diff" command. This path is scanned and then identifies the
-//  project type (API, App,.. ) from this method.
+//
+//	project type (API, App,.. ) from this method.
+//
```
Comment (npamudika) [inline import-export-cli/git/gitUtils.go:709]: Seems this change is not required. Please check.

File: import-export-cli/impl/deleteAPIProduct.go [hunk 5]
```diff
@@ -33,12 +33,13 @@ import (
 // @param accessToken : Access Token for the resource
 // @param environment : Environment where API Product needs to be located
 // @param apiProductName : Name of the API Product
+// @param apiProductVersion : Version of the API to delete
```
Comment (npamudika) [inline import-export-cli/impl/deleteAPIProduct.go:36]: *API Product

File: import-export-cli/git/gitUtils.go (hunk 1 above)
Comment (SavinduDimal) [reply import-export-cli/git/gitUtils.go]: Resolved by https://github.com/wso2/product-apim-tooling/pull/1027/commits/bf810875b56eb0bb7f3328899408af14d8d243e8

File: import-export-cli/git/gitUtils.go (hunk 3 above)
Comment (SavinduDimal) [reply import-export-cli/git/gitUtils.go]: Resolved by https://github.com/wso2/product-apim-tooling/pull/1027/commits/bf810875b56eb0bb7f3328899408af14d8d243e8

File: import-export-cli/git/gitUtils.go (hunk 4 above)
Comment (SavinduDimal) [reply import-export-cli/git/gitUtils.go]: Resolved by https://github.com/wso2/product-apim-tooling/pull/1027/commits/bf810875b56eb0bb7f3328899408af14d8d243e8

**Review by HiranyaKavishani — APPROVED** [review]

**Pull request merged**